### PR TITLE
Create dedicated simulator section

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Covers bits of #154

### CODE REVIEW

File: content/simulator/_index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Directory structure
+
+In VSCode, you will see a directory structure which looks similar to this:
+
+``` plain
+.
+├── competition-simulator
+│   ├── controllers
+│   │   ├── example_controller
+│   │   └── sr_controller
```
Comment (Dasepho) [inline content/simulator/_index.md:25]: Reference to 'sr', maybe rename?

File: content/simulator/programming/_index.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+approach to handling time. See the documentation about [simulated time](./time)
+for more details.
+
+## Other robots
+
+There is another robot in the arena, in its starting zone. By default, this robot will not move.
+
+To control the other robot, you will need to provide its code:
+
+Create the robot code in `zone-1/robot.py`. Your code at `robot.py` will be used for zone 0, though you are encouraged to move your code to `zone-0/robot.py` when running multiple robots. Robots which do not have any robot code in their `zone-x` directory will not do anything.
```
Comment (Dasepho) [inline content/simulator/programming/_index.md:40]: hasn't `robot.py' been replaced by `forklift.py' and `crane.py'? Check naming

File: content/simulator/programming/_index.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+``` plain
+.
+├── competition-simulator
+│   ├── controllers
+│   │   ├── example_controller
+│   │   └── sr_controller
```
Comment (Dasepho) [inline content/simulator/programming/_index.md:47]: reference to 'sr', rename?

File: content/simulator/programming/time.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+title: Time
+---
+
+In the simulated environment, time advances only at the pace that the simulator is run and in cooperation with the controller code (such as the code controlling your robot). As a result, using `time.time` to know how long your robot has been running for or `time.sleep` to wait for some duration will have unpredictable results, especially when run on a computer other than the one you are developing on (such as during the competition).
+
+Instead your code should explicitly signal to the simulator when it is ready for time to advance and should query for the current time from the simulator. This can be done by using the methods detailed below.
+
+{{% notice tip %}}
+If you find that the simulator freezes then this indicates that your code is busy doing something without giving the simulator a chance to run. This usually indicates that you have a loop somewhere which is expecting time to advance on its own and which should be modified to call `R.sleep` occasionally (even a very small value will allow the simulator to progress).
+{{% /notice %}}
```
Comment (Dasepho) [inline content/simulator/programming/time.md:11]: need for example snippet?

File: content/simulator/programming/time.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## Sleeping
+
+If you want to wait for something to happen within the simulation, and you can be reasonably confident that it will take a given amount of time, you can use sleep method on your robot to pause your code for a given duration.
+
+Internally this uses the simulation’s own clock and so is suitable for use in place of `time.sleep`. Note that, just as with `time.sleep`, while your code is sleeping your robot will continue any actions it was doing.
+
+```python
+# Blink the output
+R.ruggeduinos[0].digital_write(A_PIN, 1)
```
Comment (Dasepho) [inline content/simulator/programming/time.md:21]: check, `ruggeduinos` vs `arduino`, this might have been renamed?

**Review by Dasepho — COMMENTED** [review]
some naming things to check, looks like some SR naming conventions. I don't know which are overlap and which differ

File: content/simulator/programming/time.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+R.sleep(1.5)  # Sleep for a second and a half of simulation time
+R.ruggeduinos[0].digital_write(A_PIN, 0)
```
Comment (tpoomlmly) [inline content/simulator/programming/time.md:23]:
```suggestion
R.arduino.pins[8].digital_state = False
```

**Review by tpoomlmly — CHANGES_REQUESTED** [review]
We should also include docs on the IO that the simulated arduino has

File: content/simulator/_index.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+{{% notice note %}}
+Should you wish to run the simulator locally, this is fine, however will drastically limit your ability to collaborate with the rest of your team.
+{{% /notice %}}
+
+## Running simulations
+
+To ease the load on your computers, the Webots simulator will be run by us, which you can interact with through our runner interface. The runner allows you to run simulations, and view the logs and recordings afterwards.
+
+To run a simulation, click the "Run Simulation" button. The button will prevent you from running additional simulations whilst a simulation is running. Once the simulation has finished, the button will return to "Run Simulation", and an entry will be shown below it with a link to the logs and animation for your simulation run. Note that this always uses the files.
```
[the remark this answers is not in the document]
Comment (RealOrangeOne) [reply content/simulator/_index.md:14]: Unfinished sentence. Good catch!

File: content/simulator/local.md [hunk 8]
```diff
@@ -17,7 +16,9 @@ You will also need Python 3.9 installed.
 
 There are a small number of [external libraries](/api#included-libraries) which will be available to your robot code during the competition. For local development you will need to install these yourself.
 
-{{% notice warning %}} If you are installing python on Windows, you will need to add python to PATH otherwise Webots will not be able to find the python installation {{% /notice %}}
+{{% notice warning %}}
+If you are installing python on Windows, you will need to add python to `PATH` otherwise Webots will not be able to find the python installation
```
[the remark this answers is not in the document]
Comment (RealOrangeOne) [reply content/simulator/local.md:20]: The python tutorial doesn't and shouldn't have anything to do with webots

File: content/simulator/programming/_index.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+│   └─ worlds
+│       └── Arena.wbt
+├── crane.py
+└── forklift.py
+```
+
+Your code for the forklift robot should go in `forklift.py` and the crane in `crane.py`.
+
+{{% notice warning %}}
+Only your controller code will be present in the competition environment.
```
[the remark this answers is not in the document]
Comment (RealOrangeOne) [reply content/simulator/programming/_index.md:21]: Their robot code, ie not ours.

File: content/simulator/_index.md (hunk 1 above)
Comment (RealOrangeOne) [reply content/simulator/_index.md:25]: Matches how it is now, can rename once that's done. This is part of the private API so something they should never hit anyway

File: content/simulator/programming/time.md (hunk 4 above)
Comment (RealOrangeOne) [reply content/simulator/programming/time.md:11]: Putting code in notices doesn't really work, and even then I'm not sure what this snippet would look like?

File: content/simulator/local.md [hunk 10]
```diff
@@ -1,9 +1,8 @@
 ---
-title: Simulator
-weight: 5
+title: Local installation
 ---
 
-To test out your robot code, you can use our Simulator. This gives you a realistic environment to run your code, against a dummy robot, without the need for a physical arena. This simulator is currently only available on Windows.
+Should you wish to run the simulator locally, this is fine, however will drastically limit your ability to collaborate with the rest of your team.
```
[the remark this answers is not in the document]
Comment (RealOrangeOne) [reply content/simulator/local.md:5]: This intentionally wasn't a notice, it was an intro

File: content/simulator/programming/_index.md [hunk 11]
```diff
@@ -37,7 +37,7 @@ There is another robot in the arena, in its starting zone. By default, this robo
 
 To control the other robot, you will need to provide its code:
 
-Create the robot code in `zone-1/robot.py`. Your code at `robot.py` will be used for zone 0, though you are encouraged to move your code to `zone-0/robot.py` when running multiple robots. Robots which do not have any robot code in their `zone-x` directory will not do anything.
+Create the robot code in `zone-1/forklift.py` and `zone-1/crane.py`. Your original code will still be used for zone 0, though you are encouraged to move your code to `zone-0/` when running multiple robots. Robots which do not have any robot code in their `zone-x` directory will not do anything.
```
Comment (tpoomlmly) [inline content/simulator/programming/_index.md:40]: the `zone-0` folder is created by default, so their files will probably already be in there

**Comment by tpoomlmly** [comment]
Moving the simulator page broke a link on the kit page

File: content/simulator/programming/_index.md (hunk 11 above)
Comment (RealOrangeOne) [reply content/simulator/programming/_index.md:40]: I don't think it is is it? I think it just creates the in the root

File: content/simulator/programming/_index.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+On first run, the robot will execute an example program for convenience. On first run, this will be copied into `zone-0` in the same directory that `competition-simulator-<version>` is stored in:
+
+```
+.
+├── competition-simulator-<version>
+│   ├── ...
+│   └─ worlds
+│       └── Arena.wbt
+├── crane.py
```
Comment (tpoomlmly) [inline content/simulator/programming/_index.md:14]: `crane.py` and `forklift.py` are in the `zone-0` directory by default

File: content/simulator/programming/_index.md (hunk 11 above)
Comment (tpoomlmly) [reply content/simulator/programming/_index.md:40]: it created it in `zone-0` for me

File: content/simulator/programming/_index.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
+---
+title: Programming
+weight: 1
+---
+
+On first run, the robot will execute an example program for convenience. On first run, this will be copied into `zone-0` in the same directory that `competition-simulator-<version>` is stored in:
+
+```
```
Comment (tpoomlmly) [inline content/simulator/programming/_index.md:8]:
```suggestion
```plain
```

File: content/simulator/programming/_index.md (hunk 3 above)
Comment (tpoomlmly) [reply content/simulator/programming/_index.md:47]: that's how it is in the simulator atm

**Review by tpoomlmly — CHANGES_REQUESTED** [review]

**Review by tpoomlmly — DISMISSED** [review]

**Review by Montinje — APPROVED** [review]

**Pull request merged**